File: source/code/python/authentication/create_mapping_rule.py
rule_uid = AttributeStatementMappingRule(name='uid', account_attributes=['username'])
rule_mail = AttributeStatementMappingRule(name='mail', account_attributes=['email'])
rule_location = AttributeStatementMappingRule(name='location', account_attributes=['customData.location'])

asmr = directory.provider.attribute_statement_mapping_rules
asmr.items = [rule_uid, rule_mail, rule_location]
asmr.save()

File: source/code/nodejs/authentication/create_mapping_rule.js
directory.getProvider({ expand: 'attributeStatementMappingRules' }, function (err, provider) {
  if (err) {
    return console.error(err);
  }

  provider.attributeStatementMappingRules.items.push(newAttributeMappingRule);

  provider.save(function(err){
    if (err) {
      return console.error(err);
    }

    console.log('Attribute mapping rules have been saved');
  });

});